### DISCUSSION TITLE
How to bind to StringFormat ？

### DISCUSSION THREAD

**Discussion by nasterhe** [AvaloniaUI/Avalonia#17728] [Q&A] [answered]
My data display format changes dynamically, so I need to bind StringFormat, but the code below is incorrect.
` <DataGridTextColumn Header="Number" Binding="{Binding Number,StringFormat=#mainGrid.((vmt:ThisViewModel)DataContext).StringFormat}" x:DataType="vmt:CThisHelper" />`
The error message:Unable to find suitable setter or adder for property StringFormat of type Avalonia.Markup:Avalonia.Data.BindingBase for argument Avalonia.Markup.Xaml:Avalonia.Markup.Xaml.MarkupExtensions.CompiledBindingExtension, available setter parameter lists are:
Can anyone help me, thanks.

**Comment by stevemonaco** [comment, marked as answer]
`StringFormat` isn't a parameter that can be binded. It's a normal CLR property.

I would suggest using a `MultiBinding` here with the format and number as parameters. Then write a converter to format the text. You can look at https://github.com/AvaloniaUI/Avalonia/blob/4c428d073d28afc1196aff7c60168db62754ef05/src/Avalonia.Controls/Converters/StringFormatConverter.cs#L13-L29 for reference (and it's basically 100% of what you need for the converter side), but it's not recommended for direct usage.
